packages/core/android/build.gradle

```diff
@@ -84,6 +84,16 @@ def getExtOrIntegerDefault(name) {
 android {
   compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
   buildToolsVersion getExtOrDefault('buildToolsVersion')
+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+    namespace = "com.datadog.reactnative"
+  }
+  if (agpVersion.tokenize('.')[0].toInteger() >= 8) {
+    buildFeatures {
+      buildConfig = true
+    }
+  }
```

Comment:
I'm not quite sure it will work (but who knows, it is Groovy). If types are statically checked, then this build script won't compile with AGP 6.x even if new properties are wrapped in `if`. But if things are dynamic (maybe because of Groovy), it will be fine. Does you change work fine if you are running it in the context of AGP 6.x and 7.x?

Reply:
I tested with AGP 4.2.2 (no RN version uses AGP 5.x or 6.x, but can test these as well), and it worked fine.
Also I'm a bit surprised that even if I remove these `if` conditions everything builds as well 🤷‍♂️ 

Still, adding these `if` conditions is what's done by most RN libraries and I might be missing something here. I don't want to spend too much time on this so I think it's ok to go with the safest approach even if it adds a bit of useless code.